index.dd

```diff
@@ -123,6 +123,21 @@ void main()
 }
 ----
 )
+$(EXTRA_EXAMPLE_RUNNABLE Print hex dump,
+----
+void main()
+{
+    import std.algorithm, std.stdio, std.file;
+    enum cols = 16;
+    thisExePath.File("rb").byChunk(cols).each!(chunk =>
+        writefln!"%(%02X %)%*s  %s"(
```

Comment:
Not sure that would be much of an improvement.

- C doesn't have ranges or range formatting syntax in printf
- The fmt!formatString is pretty similar to Rust - C programmers interested in D will almost certainly also be interested in Rust
- Compile-time format string parsing is awesome - D didn't have support for it for a long time, which meant mistakes in the format string weren't caught until runtime.